pyiron/lammps/structure.py

```diff
@@ -402,10 +402,10 @@ def structure_full(self):
 
         # extract electric charges from potential file
         q_dict = {}
-        for el in self._structure.get_species_symbols():
-            q_dict[el] = float(self.potential.get("set group {} charge".format(el)))
-        species_translate_list = [self.potential.get("group {} type".format(el.Abbreviation))
-                                  for el in self.structure.species]
+        species_translate_list = []
+        for el in self.structure.get_species_symbols():
+            q_dict[el] = self.potential.get_charge(el)
+            species_translate_list.append(self.potential.get_element_id(el))
```

Comment:
For these lines, you need to change them to

```
        species_translate_list = []
        for el in self.structure.species:
            q_dict[el.Abbreviation] = self.potential.get_charge(el.Abbreviation)
            species_translate_list.append(self.potential.get_element_id(el.Abbreviation))
```
This is because, `self.structure.get_species_symbols()` and `self.structure.species` do not return the species in the same order. `species_translate_list` should follow the order of `self.structure.species`